crates/ruff_workspace/src/configuration.rs

```diff
@@ -154,130 +149,140 @@ impl Configuration {
-                .unwrap_or_default(),
-            flake8_tidy_imports: self
-                .flake8_tidy_imports
-                .map(Flake8TidyImportsOptions::into_settings)
-                .unwrap_or_default(),
-            flake8_type_checking: self
-                .flake8_type_checking
-                .map(Flake8TypeCheckingOptions::into_settings)
-                .unwrap_or_default(),
-            flake8_unused_arguments: self
-                .flake8_unused_arguments
-                .map(Flake8UnusedArgumentsOptions::into_settings)
-                .unwrap_or_default(),
-            flake8_gettext: self
-                .flake8_gettext
-                .map(Flake8GetTextOptions::into_settings)
-                .unwrap_or_default(),
-            isort: self
-                .isort
-                .map(IsortOptions::try_into_settings)
-                .transpose()?
-                .unwrap_or_default(),
-            mccabe: self
-                .mccabe
-                .map(McCabeOptions::into_settings)
-                .unwrap_or_default(),
-            pep8_naming: self
-                .pep8_naming
-                .map(Pep8NamingOptions::try_into_settings)
-                .transpose()?
-                .unwrap_or_default(),
-            pycodestyle: self
-                .pycodestyle
-                .map(PycodestyleOptions::into_settings)
-                .unwrap_or_default(),
-            pydocstyle: self
-                .pydocstyle
-                .map(PydocstyleOptions::into_settings)
-                .unwrap_or_default(),
-            pyflakes: self
-                .pyflakes
-                .map(PyflakesOptions::into_settings)
-                .unwrap_or_default(),
-            pylint: self
-                .pylint
-                .map(PylintOptions::into_settings)
-                .unwrap_or_default(),
-            pyupgrade: self
-                .pyupgrade
-                .map(PyUpgradeOptions::into_settings)
-                .unwrap_or_default(),
+            linter: LinterSettings {
+                target_version,
+                project_root: project_root.to_path_buf(),
+                rules,
+                allowed_confusables: self
+                    .allowed_confusables
+                    .map(FxHashSet::from_iter)
+                    .unwrap_or_default(),
+                builtins: self.builtins.unwrap_or_default(),
+                dummy_variable_rgx: self
+                    .dummy_variable_rgx
+                    .unwrap_or_else(|| DUMMY_VARIABLE_RGX.clone()),
+                external: FxHashSet::from_iter(self.external.unwrap_or_default()),
+                ignore_init_module_imports: self.ignore_init_module_imports.unwrap_or_default(),
+                line_length: self.line_length.unwrap_or_default(),
+                tab_size: self.tab_size.unwrap_or_default(),
+                namespace_packages: self.namespace_packages.unwrap_or_default(),
```

Comment:
I feel like some of these aren't linter-specific, like `namespace_packages` and `src` -- they would be generally applicable to any tool that does module / file resolution. But it's not important for now.

Reply:
That's true, but the `Settings` are the resolved view for a tool. My current, not so nice approach, is to duplicate settings that apply to multiple tools when resolving the `Configuration`. Meaning, the `Configuration` applies` the right inheritance rules and, as a result, may end up initializing common fields identically. 

I don't think this is a performance problem because we only do this once per setting and settings tend to not have very large structures.